src/docs/asciidoc/distributed_sql.adoc

```diff
@@ -0,0 +1,211 @@
+
+
+== Distributed SQL
+
+Distributed SQL allows you to access data stored in the Hazelcast cluster
+declaratively using SQL.
+
+=== Beta Status
+
+The distributed SQL is currently in beta. The compatibility between versions
+is not guaranteed. API might change between versions without notice.
+
+=== How Distributed SQL works
+
+When an SQL statement is submitted for execution, the engine parses and
+optimizes it using Apache Calcite. The result is an executable plan that
+is cached and reused by the subsequent executions of the same statement.
+
+The plan contains a tree of query fragments. A query fragment is a tree of
+operators that could be executed on a single member independently. Child
+fragments supply data to parent fragments, possibly through a network, until
+the root fragment is reached. The root fragment returns query results to a
+user.
+
+Only `IMap` objects are supported at the moment.
+
+=== Querying IMap
+
+==== Names
+
+The engine exposes `IMap` objects as tables in the predefined `partitioned`
```

Comment:
Let's not use the term `engine`. It sounds very low-level. Perhaps we can put a short paragraph saying something like "All SQL related operations are exposed via instance.getSqlService" and then talk about "SQL service exposes..." or maybe do passive "IMap objects are exposed as tables in ... "
Also, I believe that noone will actually use the `partitioned` prefix (or for sure not in the very beginning), so let's change the structure of the paragraph a bit:
* IMap objects are exposed as table in a case sensitive manner. For example ..."
* All IMaps are exposed in the predefined `partitioned` schema. Therefore you can use <example>

Reply:
Renamed "engine" to "SQL service". 

Regarding schema - for IMDG this will be important in the future when we add support for replicated maps, that will be located in the `replicated` schema, and when we have JDBC. For Jet users, this note is important right away, because they will possibly remap objects to different schemas with DML/DDL. Moreover, "schema" is a well-known term in the SQL world, and users expect that every object resides in some container. Therefore, I would change that.